### PR TITLE
[Verification] Adds metrics collector and tracer to requester and fetcher engines

### PR DESCRIPTION
**Pull request by yhassanzadeh13**
This PR adds metrics collector and tracer to the new `requester` and `fetcher` engines of verification node. Note that **this PR does not change the logic of engines**, and most of the changes in the engine codes are **code relocation** to encapsulate the tracing and metrics.

### CODE REVIEW

File: engine/verification/test/assignerfetcher_test.go [hunk 1]
```diff
@@ -113,173 +98,15 @@ func TestVerificationHappyPath(t *testing.T) {
...
+				tc.staked,
+				tc.blockCount,
+				tc.eventRepetition,
+				collector,
+				collector,
+				tc.ops...)
 		})
 	}
 }
-
```
Comment (yhassanzadeh13) [inline engine/verification/test/assignerfetcher_test.go:135]: moving this test helper to `helper.go` file.

File: engine/verification/utils/unittest/helper.go [hunk 2]
```diff
@@ -545,3 +549,185 @@ func MockLastSealedHeight(state *mockprotocol.State, height uint64) {
 	state.On("Sealed").Return(snapshot)
 	snapshot.On("Head").Return(&header, nil)
 }
+
+func NewVerificationHappyPathTest(t *testing.T,
```
Comment (yhassanzadeh13) [inline engine/verification/utils/unittest/helper.go:553]: This is just moving a helper to here, no new logic has been developed.

File: module/metrics.go [hunk 3]
```diff
@@ -159,17 +159,13 @@ type ConsensusMetrics interface {
 }
 
 type VerificationMetrics interface {
-	// Finder Engine
-	//
+	// TODO: remove this event handlers once we have new architecture in place.
```
Comment (yhassanzadeh13) [inline module/metrics.go:162]: We plan to _group these methods_ into smaller child interfaces each for a separate engine: https://github.com/dapperlabs/flow-go/issues/5534

File: engine/verification/test/assignerfetcher_test.go [hunk 4]
```diff
@@ -44,7 +29,7 @@ import (
 func TestVerificationHappyPath(t *testing.T) {
 	testcases := []struct {
 		blockCount      int
-		opts            []vertestutils.CompleteExecutionReceiptBuilderOpt
+		ops             []vertestutils.CompleteExecutionReceiptBuilderOpt
```
Comment (jordanschalm) [inline engine/verification/test/assignerfetcher_test.go:32]:
```suggestion
		opts             []vertestutils.CompleteExecutionReceiptBuilderOpt
```
Maybe overwriting a commit from previous PR? Pretty sure I've made this suggestion previously 😅

**Review by jordanschalm — APPROVED** [review]
Nice work

**Review by zhangchiqing — APPROVED** [review]
Looks good

File: engine/verification/fetcher/engine.go [hunk 5]
```diff
@@ -186,64 +225,141 @@ func (e *Engine) HandleChunkDataPack(originID flow.Identifier, chunkDataPack *fl
...
 
+	ctx := opentracing.ContextWithSpan(e.unit.Ctx(), span)
+	var fatalErr, nonFatalErr error
+	var processed bool
+	e.tracer.WithSpanFromContext(ctx, trace.VERFetcherHandleChunkDataPack, func() {
+		// make sure the chunk data pack is valid
+		err := e.validateChunkDataPackWithTracing(ctx, status.ChunkIndex, originID, chunkDataPack, collection, status.ExecutionResult)
+		if err != nil {
+			// TODO: this can be due to a byzantine behavio
+			nonFatalErr = fmt.Errorf("could not validate chunk data pack: %w", err)
```
[the remark this answers is not in the document]
Comment (yhassanzadeh13) [reply engine/verification/fetcher/engine.go]: Fixed in https://github.com/onflow/flow-go/pull/707/commits/5e722892fe2922f8da0ebe007ffbd520ef71d9b4

File: engine/verification/fetcher/engine.go [hunk 6]
```diff
@@ -186,64 +225,141 @@ func (e *Engine) HandleChunkDataPack(originID flow.Identifier, chunkDataPack *fl
...
 		Hex("result_id", logging.ID(resultID)).
 		Uint64("chunk_index", status.ChunkIndex).
 		Logger()
 
-	// make sure the chunk data pack is valid
-	err := e.validateChunkDataPack(chunk, originID, chunkDataPack, collection, status.ExecutionResult)
-	if err != nil {
-		// TODO: this can be due to a byzantine behavior
-		lg.Error().Err(err).Msg("could not validate chunk data pack")
+	processed, nonFatalErr, fatalErr := e.handleChunkDataPackWithTracing(originID, status, chunkDataPack, collection)
```
[the remark this answers is not in the document]
Comment (yhassanzadeh13) [reply engine/verification/fetcher/engine.go]: Fixed in https://github.com/onflow/flow-go/pull/707/commits/5e722892fe2922f8da0ebe007ffbd520ef71d9b4

**Pull request merged**